perfkitbenchmarker/linux_benchmarks/object_storage_service_benchmark.py

```diff
@@ -293,9 +326,145 @@ def _MakeSwiftCommandPrefix(auth_url, tenant_name, username, password):
   return ' '.join(options)
 
 
+def _ParseMultiStreamResults(raw_result, metadata=None):
+  """Read results from the api_multistream worker process.
+
+  Args:
+    raw_result: string. The stdout of the worker process.
+    metadata: dict. Base sample metadata
+
+  Returns:
+    A list of sample.Sample objects.
+  """
+
+  if metadata is None:
+    metadata = {}
+
+  records = json.loads(raw_result)
+  results = []
+
+  for record in records:
+    metric = 'Multi-stream %s KB %s' % (record['size'], record['operation'])
+    md = metadata.copy()
```

Comment:
The original metadata contains fields that are common to all of the samples, but each sample will have a different stream_num and object_size. So I make a copy of the common metadata dictionary and update it with the unique fields for each sample.